docs/proposals/scheduling/20230314-gpu-topology-aware-scheduling.md

```diff
@@ -0,0 +1,275 @@
+---
+title: GPU topology-aware scheduling
+authors:
+  - "@happy2048"
+reviewers:
+  - "@eahydra"
+  - "@hormes"
+  - "@yihuifeng"
+  - "@honpey"
+  - "@zwzhang0107"
+  - "@jasonliu747"
+creation-date: 2023-03-14
+
+---
+
+
+# GPU topology-aware scheduling
+
+## Table of Contents
+<!--ts-->
+
+* [GPU topology-aware scheduling](#gpu-topology-aware-scheduling)
+   * [Table of Contents](#table-of-contents)
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals/Future Work](#non-goalsfuture-work)
+   * [Proposal](#proposal)
+      * [User stories](#user-stories)
+         * [Story 1](#story-1)
+         * [Story 2](#story-2)
+      * [Implementation Details](#implementation-details)
+         * [main steps](#main-steps)
+         * [How to identify pods are in the same group](#how-to-identify-pods-are-in-the-same-group)
+         * [GPU topology resource reporting](#gpu-topology-resource-reporting)
+         * [Node Selection](#node-selection)
+         * [Pick GPUs of Node](#pick-gpus-of-node)
+         * [Record the allocation information to the pod annotation](#record-the-allocation-information-to-the-pod-annotation)
+         * [Container environment variable assignment](#container-environment-variable-assignment)
+         * [Works with the Gang plugin](#works-with-the-gang-plugin)
+   * [Unsolved Problems](#unsolved-problems)
+
+<!--te-->
+
+
+## Summary
+In Distributed Deep Learning Job, each worker for the training job may involve data exchange and other operations. The bandwidth between GPU cards will affect the training time of the training job. Although the k8s native scheduler can allocate GPU cards to the workers of the training job, bandwidth between GPU cards is not considered; this proposal will provide a scheduling plugin to consider the bandwidth between a group of GPU cards when allocating a group of GPU cards to pods.
+## Motivation
+NVIDIA Collective Communication Library (NCCL) is a Magnum IO library provided by NVIDIA, which can realize GPU-accelerated collective operations. NCCL is topology-aware (automatically perceives the connection type between GPU cards, no manual configuration is required) and is optimized to pass PCIe, NVLink, Ethernet, and InfiniBand interconnects enable high bandwidth and low latency. In the deep learning distributed training job, the distributed training framework (Pytorch, MPI) combined with the NCCL library can achieve the acceleration effect. The NCCL library can perceive the connection between the GPU cards. Different connection types have different bandwidths. The size of the bandwidth affects the training time of the training job.
+
+The following is a matrix describing the bandwidth between 8 GPU cards on a node, and the unit of value is GB/s:
```

Comment:
Strictly speaking， not all 8-card gpu type support nvlink, should we annouce some example gpu models like v100\100?(or 1080Ti not support)?